logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -51,12 +51,12 @@ def fetch_or_store(namespaces, key, default_value = nil)
       # BUT. If the value is not present in the `@fast_lookup` the value will be inserted and we assume that we don't
       # have it in the `@metric_store` for structured search so we add it there too.
 
-      value = @fast_lookup.get(namespaces.dup << key)
+      value = @fast_lookup.get((namespaces.dup << key).hash)
```

Comment:
@jakelandis One more thought: even if the individual collision likelihood is small, this is a repeat run game with every run of LS for every user ...

=> at 30 values, you have a chance of a collision of 5% with 100k LS runs ... so this is effectively guaranteed to affect people randomly